src/pubsub/src/subscriber/stream.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::stub::Stub;
+use crate::google::pubsub::v1::StreamingPullRequest;
+use crate::{Error, Result};
+use gax::options::RequestOptions;
+use tokio::sync::mpsc::{Sender, channel};
+
+/// Open a stream for the `StreamingPull` RPC.
+///
+/// This returns the stream and a Sender for feeding the stream writes.
+pub(crate) async fn open_stream<T>(
+    inner: T,
+    initial_req: StreamingPullRequest,
+) -> Result<(<T as Stub>::Stream, Sender<StreamingPullRequest>)>
+where
+    T: Stub,
+{
+    // The only writes we perform are keepalives, which are sent so infrequently
+    // that we don't fear any back pressure on this channel.
+    let (request_tx, request_rx) = channel(1);
```

Comment:
nit: light preference to keep mpsc::channel and mspc::Sender to make it really easy to see what kind of channel and sender we are using.

Reply:
mehh I don't really agree, but don't feel strongly, so done. :shrug: 

It doesn't help distinguish from [`std::sync::mpsc::channel`](https://doc.rust-lang.org/stable/std/sync/mpsc/fn.channel.html) (which has confused me in a review before, lol)

And we often have to look at the use statements to know which `Result`/`Error` type the code is talking about.